commit b181e4a261fe63dff22250e504f324b66bab5a7e
Author: bertin <macportsuser@macports.org>
Date:   Fri May 23 00:02:10 2025 +0200

    /opt/local/site-ports/sysutils/MacPorts/files/lamarck/patch-stubports-no-pre-or-post.diff

diff --git a/src/port1.0/portutil.tcl b/src/port1.0/portutil.tcl
index 7cb099ced..fafc74297 100644
--- a/src/port1.0/portutil.tcl
+++ b/src/port1.0/portutil.tcl
@@ -1553,7 +1553,8 @@ proc target_run {ditem} {
 
                 # For {} blocks in the Portfile, export DEVELOPER_DIR to prevent Xcode binaries if shouldn't be used
                 set env(DEVELOPER_DIR) [option configure.developer_dir]
-                if {$result == 0} {
+                set is_stub [string match *setup_stub_linux [info proc stub::setup_stub_linux]]
+                if {$result == 0 && $is_stub == 0} {
                     foreach pre [ditem_key $ditem pre] {
                         ui_debug "Executing $pre"
                         set result [catch {$pre $targetname} errstr]
@@ -1572,7 +1573,7 @@ proc target_run {ditem} {
                     set errinfo $::errorInfo
                 }
 
-                if {$result == 0} {
+                if {$result == 0 && $is_stub == 0} {
                     foreach post [ditem_key $ditem post] {
                         ui_debug "Executing $post"
                         set result [catch {$post $targetname} errstr]
